bin/block_generator/block_generator.py

```diff
@@ -33,48 +34,50 @@ def __exit__(self, *exit_args):
 
 # Interface for usage of nil_cli
 class NilCLI:
-    def __init__(self, config_path: str):
-        self.config_path = config_path
+    def __init__(self):
+        pass
 
     def run_cli(self, args: list[str], pattern=None) -> str:
-        full_arg_list = ["nil_cli", "-c", self.config_path] + args
+        full_arg_list = ["nil_cli"]
+        full_arg_list += args
         logging.info("Launching " + " ".join(full_arg_list))
         p = subprocess.run(full_arg_list, capture_output=True)
-        logging.debug(p.stderr)
+        logging.debug(p.stdout)
         if p.returncode != 0:
             raise RuntimeError(p.stderr)
-        output = p.stderr.decode()
+        output = p.stdout.decode()
         if pattern is None:
             return output
         a = re.search(pattern, output)
         if a is None:
-            raise RuntimeError(p.stderr)
+            return None
         return a.group(1)
 
-    def init_config(self, endpoint: str):
-        self.run_cli(["config", "init"])
-        self.run_cli(["config", "set", "rpc_endpoint", endpoint])
-
-    def keygen(self) -> str:
-        return self.run_cli(["keygen", "new"])
+    def wallet_is_exist(self) -> str:
+        wallet_pattern = "Wallet address: (0x[0-9a-fA-F]+)"
+        return self.run_cli(["wallet", "info"], pattern=wallet_pattern)
 
     def new_wallet(self) -> str:
-        return self.run_cli(["wallet", "new"])
+        address = self.wallet_is_exist()
+        if address is not None:
+            return address
+        wallet_pattern = "New wallet address: (0x[0-9a-fA-F]+)"
+        return self.run_cli(["wallet", "new"], pattern=wallet_pattern)
 
     def deploy_contract(self, bytecode_file: str, abi_file: str) -> str:
         contract_pattern = "Contract address: (0x[0-9a-f]+)"
         deploy_args = ["wallet", "deploy", bytecode_file]
         deploy_args += ["--abi", abi_file]
-        salt = random.getrandbits(64)
-        deploy_args += ["--salt", str(salt)]
```

Comment:
Different salts allow us to deploy multiple instances of a contract with the same code, otherwise `nil_cli` returns an error. For example, I was unable to call the script twice with the same config file (not sure about the latest version). Are you sure that this can't be useful?

Reply:
useless for single wallet, but not sure about multiple wallets case, I'll recheck with the latest version